src/librustc_codegen_ssa/base.rs

```diff
@@ -508,6 +509,48 @@ fn get_argc_argv<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
 
 pub const CODEGEN_WORKER_ID: usize = ::std::usize::MAX;
 
+/// Check for the `#[rustc_error]` annotation, which forces an
+/// error in codegen. This is used to write compile-fail tests
+/// that actually test that compilation succeeds without
+/// reporting an error.
+pub fn check_for_rustc_errors_attr(tcx: TyCtxt<'_>) {
+    let def_id = match tcx.entry_fn(LOCAL_CRATE) {
+        Some((def_id, _)) => def_id,
+        _ => return,
+    };
+
+    const DELAY_SPAN_BUG_TRIGGERED: &str =
+        "delayed span bug triggered by #[rustc_error(delay_span_bug_from_inside_query)]";
+
+    let attrs = &*tcx.get_attrs(def_id);
+    let attrs = attrs.iter().filter(|attr| attr.check_name(sym::rustc_error));
+    for attr in attrs {
+        match attr.meta_item_list() {
+            // Check if there is a `#[rustc_error(delayed)]`.
+            Some(list)
+                if list.iter().any(|list_item| {
+                    matches!(
+                        list_item.ident().map(|i| i.name),
+                        Some(sym::delay_span_bug_from_inside_query)
```

Comment:
Isn't the whole point of this to test `delay_span_bug` interactions with queries?

Reply:
I'm not sure I follow. What do you mean?

Reply:
@eddyb ^^?

Reply:
Oh, GitHub didn't point me to this.

The name suggests the `delay_span_bug` would be triggered from inside a query, that's why there's a query that gets provided, but you removed all that.